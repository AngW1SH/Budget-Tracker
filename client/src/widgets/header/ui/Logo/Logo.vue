<script setup lang="ts"></script>

<template>
  <router-link to="/">
    <div class="flex items-center">
      <div class="h-4 w-4 rounded-full bg-purple-800 mr-3 mt-[3px]"></div>
      <h1 class="text-2xl font-medium">Budget Tracker</h1>
    </div>
  </router-link>
</template>
