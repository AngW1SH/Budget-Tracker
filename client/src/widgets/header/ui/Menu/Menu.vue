<script setup lang="ts"></script>

<template>
  <ul class="flex text-gray-500 font-semibold">
    <li class="mr-12">Categories</li>
    <li class="mr-12">About</li>
    <li>Pricing</li>
  </ul>
</template>
