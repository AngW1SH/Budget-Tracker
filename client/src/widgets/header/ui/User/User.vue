<script setup lang="ts">
import { SignupButton, LoginButton } from "@/features/auth";
</script>

<template>
  <div>
    <LoginButton class="mr-6" />
    <SignupButton />
  </div>
</template>
