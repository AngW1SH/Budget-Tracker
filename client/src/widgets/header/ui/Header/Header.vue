<script setup lang="ts">
import { Container } from "@/shared/ui";
import Logo from "../Logo/Logo.vue";
import Menu from "../Menu/Menu.vue";
import User from "../User/User.vue";
</script>

<template>
  <div class="py-6 shadow-center-md">
    <Container class="flex items-center">
      <Logo class="mr-14" />
      <Menu class="mt-[0.4rem]" />
      <User class="ml-auto" />
    </Container>
  </div>
</template>
