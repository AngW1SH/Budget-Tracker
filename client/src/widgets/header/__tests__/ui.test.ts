import { describe, expect, test } from "vitest";
import { mount } from "@vue/test-utils";
import Header from "../ui/Header/Header.vue";
import Logo from "../ui/Logo/Logo.vue";
import Menu from "../ui/Menu/Menu.vue";
import User from "../ui/User/User.vue";

describe("Header Widget UI", () => {
  test("contains UI-elements", () => {
    const header = mount(Header);
    expect(header.findComponent(Logo).exists()).toBe(true);
    expect(header.findComponent(Menu).exists()).toBe(true);
    expect(header.findComponent(User).exists()).toBe(true);
  });
  test("snapshot UI testing", () => {
    const header = mount(Header);
    expect(header.element).toMatchSnapshot();
  });
});
