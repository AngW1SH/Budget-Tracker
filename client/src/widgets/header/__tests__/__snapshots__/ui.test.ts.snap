// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Widget UI > snapshot UI testing 1`] = `
<div
  class="py-6 shadow-md"
>
  <div
    class="w-8/12 mx-auto flex items-center"
  >
    
    <router-link
      class="mr-14"
      to="/"
    >
      <div
        class="flex items-center"
      >
        <div
          class="h-4 w-4 rounded-full bg-purple-800 mr-3 mt-[3px]"
        />
        <h1
          class="text-2xl font-medium"
        >
          Budget Tracker
        </h1>
      </div>
    </router-link>
    <ul
      class="flex text-gray-500 font-semibold mt-[0.4rem]"
    >
      <li
        class="mr-12"
      >
        <router-link
          to="/categories"
        >
          Categories
        </router-link>
      </li>
      <li
        class="mr-12"
      >
        <router-link
          to="/about"
        >
          About
        </router-link>
      </li>
      <li>
        <router-link
          to="/pricing"
        >
          Pricing
        </router-link>
      </li>
    </ul>
    <div
      class="ml-auto"
    >
      <router-link
        to="/login"
      >
        <button
          class="text-gray-500 font-semibold mr-6"
        >
          Login
        </button>
      </router-link>
      <router-link
        to="/login"
      >
        <button
          class="text-center text-white font-semibold bg-purple-800 py-[0.675rem] px-5 rounded-md"
        >
          
          Sign Up
          
        </button>
      </router-link>
    </div>
    
  </div>
</div>
`;
